server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1498,86 +1248,46 @@ public void deleteLogLevel(@PathParam("owner_key") String ownerKey) {
-    })
     public List<PoolDTO> listPools(
-        @PathParam("owner_key") @Verify(value = Owner.class, subResource = SubResource.POOLS) String ownerKey,
-        @QueryParam("consumer") String consumerUuid,
-        @QueryParam("activation_key") String activationKeyName,
-        @QueryParam("product") String productId,
-        @QueryParam("subscription") String subscriptionId,
-        @ApiParam("Include pools that are not suited to the unit's facts.")
-        @QueryParam("listall") @DefaultValue("false") boolean listAll,
-        @ApiParam("Date to use as current time for lookup criteria. Defaults" +
-                " to current date if not specified.")
-        @QueryParam("activeon") @DefaultValue(DateFormat.NOW) @DateFormat Date activeOn,
-        @ApiParam("Find pools matching the given pattern in a variety of fields;" +
-                " * and ? wildcards are supported; may be specified multiple times")
-        @QueryParam("matches") List<String> matches,
-        @ApiParam("The attributes to return based on the specified types.")
-        @QueryParam("attribute") List<KeyValueParamDTO> attrFilters,
-        @ApiParam("When set to true, it will add future dated pools to the result, " +
-                "based on the activeon date.")
-        @QueryParam("add_future") @DefaultValue("false") boolean addFuture,
-        @ApiParam("When set to true, it will return only future dated pools to the result, " +
-                "based on the activeon date.")
-        @QueryParam("only_future") @DefaultValue("false") boolean onlyFuture,
-        @ApiParam("Will only return pools with a start date after the supplied date. " +
-                "Overrides the activeOn date.")
-        @QueryParam("after") @DateFormat Date after,
-        @ApiParam("One or more pool IDs to use to filter the output; only pools with IDs matching " +
-                "those provided will be returned; may be specified multiple times")
-        @QueryParam("poolid") List<String> poolIds,
-        @Context Principal principal,
-        @Context PageRequest pageRequest) {
+        @Verify(value = Owner.class, subResource = SubResource.POOLS) String ownerKey,
+        String consumerUuid,
+        String activationKeyName,
+        String productId,
+        String subscriptionId,
+        Boolean listAll,
+        OffsetDateTime activeOn,
+        List<String> matches,
+        List<KeyValueParamDTO> attrFilters,
+        Boolean addFuture,
+        Boolean onlyFuture,
+        OffsetDateTime after,
+        List<String> poolIds) {
+
+        Principal principal = this.principalProvider.get();
+        PageRequest pageRequest = ResteasyContext.getContextData(PageRequest.class);
 
         Owner owner = findOwnerByKey(ownerKey);
 
+        // set default to current date
+        Date activeOnDate = activeOn != null ? Util.toDate(activeOn) : new Date();
+        Date afterDate = after != null ? Util.toDate(after) : null;
```

Comment:
For this line, you don't need to check for null. `Util.toDate()` will just return null of the arg is null

Reply:
Done.